packages/jest-haste-map/src/index.js

```diff
@@ -0,0 +1,268 @@
+ /**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const H = require('./constants');
+
+const crypto = require('crypto');
+const denodeify = require('denodeify');
+const execSync = require('child_process').execSync;
+const fs = require('graceful-fs');
+const getPlatformExtension = require('./lib/getPlatformExtension');
+const nodeCrawl = require('./crawlers/node');
+const os = require('os');
+const path = require('./fastpath');
+const watchmanCrawl = require('./crawlers/watchman');
+const worker = require('./worker');
+const workerFarm = require('worker-farm');
+
+const NODE_MODULES = path.sep + 'node_modules' + path.sep;
+const VERSION = require('../package.json').version;
+
+const canUseWatchman = (() => {
+  try {
+    execSync('watchman version', {stdio: 'ignore'});
+    return true;
+  } catch (e) {}
+  return false;
+})();
+
+class HasteMap {
+
+  constructor(options) {
+    this._options = {
```

Comment:
why not `Object.assign`?

```
this._options = Object.assign({
  cacheDirectory: os.tmpdir(),
  mocksPattern: options.mocksPattern
    ? new RegExp(options.mocksPattern)
    : null,
  useWatchMan: true,
};
```

---

also - shouldn't this be `os.tmpdir()` instead of `os.tmpDir()`?